crates/solver/src/settlement_submission/submitter/public_mempool_api.rs

```diff
@@ -56,24 +56,31 @@ impl TransactionSubmitting for PublicMempoolApi {
                             node.web3.eth().send_raw_transaction(bytes.0.into()).await
                         }
                     };
+                    let can_broadcast = node.can_broadcast();
 
-                    (label, result)
+                    (label, result, can_broadcast)
                 }
                 .boxed()
             })
             .collect::<Vec<_>>();
 
         let mut errors = vec![];
         loop {
-            let ((label, result), _, rest) = futures::future::select_all(futures).await;
+            let ((label, result, can_broadcast), _, rest) =
+                futures::future::select_all(futures).await;
             match result {
                 Ok(tx_hash) => {
                     super::track_submission_success(&label, true);
-                    tracing::debug!(%label, "created transaction with hash: {:?}", tx_hash);
-                    return Ok(TransactionHandle {
-                        tx_hash,
-                        handle: tx_hash,
-                    });
+                    if can_broadcast {
+                        tracing::debug!(%label, "created transaction with hash: {:?}", tx_hash);
+                        return Ok(TransactionHandle {
+                            tx_hash,
+                            handle: tx_hash,
+                        });
+                    } else {
+                        tracing::debug!(%label, "non-broadcasting node reports transaction creation with hash: {:?}", tx_hash);
+                        futures = rest;
```

Comment:
nit: At this point it seems less error prone to move the `futures = rest;` to after the `match` statement and remove it from the `Err` case (so we can't do it twice). That way it communicates this always has to happen unless we take an early exit.

Reply:
Very true, while implementing it I spotted a bug in these changes that could cause a panic if the last node in the loop was a notification node: we wouldn't exit the loop if the vector was empty, and `future::select_all` panics if the input is an empty vector.